src/app/projects/components/wrangler-list-input/wrangler-list-input.component.ts

```diff
@@ -76,27 +71,38 @@ export class WranglerListInputComponent extends BaseInputComponent implements On
     return wrangler ? wrangler.name : '';
   }
 
-  private getWranglerNames(): Observable<string[]> {
-    return this.ingestService.getWranglers().pipe(
-      map(wranglers => wranglers?.map(wrangler => wrangler.name)));
-  }
-
-  private setDefaultWrangler() {
-    // TODO set current account as default wrangler
-  }
-
   private onSearchValueChanged(value: string | Account): Observable<Account[]> {
     const searchText = typeof value === 'string' ? value.toLowerCase() : value.name ? value.name.toLowerCase() : '';
-    return this._filter(searchText);
+    const emptyAccount = new Account({id: null, providerReference: null, name: 'Unassigned'});
+    return this.filterWranglers(searchText).pipe(
+      tap(found_wranglers => found_wranglers.unshift(emptyAccount))
+    );
+  }
+
+  private filterWranglers(wranglerName: string): Observable<Account[]> {
+    const lowerName = wranglerName.toLowerCase();
+    return this.wranglers$.pipe(map(wranglers => wranglers.filter(wrangler => wrangler.name.toLowerCase().includes(lowerName))));
   }
 
-  private _filter(name_or_id: string): Observable<Account[]> {
-    const filterValue = name_or_id.toLowerCase();
+  private findWrangler(accountId: string): Observable<Account> {
     return this.wranglers$.pipe(
-      map(wranglers => wranglers.filter(wrangler => {
-          return wrangler.name.toLowerCase().indexOf(filterValue) === 0 || wrangler.id === name_or_id;
-        })
-      )
+      map(allWranglers => allWranglers.filter(wrangler => wrangler.id === accountId)),
+      map(matchingWranglers => {
```

Comment:
I find the contents here complex, due to high indentation levels and conditional nesting..
Have you considered adding a filter step before the map to weed out the case of `matchingWranglers.length > 0`

Reply:
With a filter we loose the opportunity to raise the warnings to tell the user what was broken and why.

Reply:
Although if we were willing to change the functionality so that in the case of multiple matching wranglers found we don't populate the dropdown at all we could have this slight simplification:
``` src/app/metadata-schema-form/metadata-form/metadata-form.component.ts
if (matchingWranglers.length === 1) {
  return matchingWranglers[0];
}
if (matchingWranglers.length > 1) {
  let names: string[] = matchingWranglers.map(wrangler => wrangler.name ? wrangler.name : 'Missing Name');
  this.alertService.warn(
    'Multiple Wranglers Match Account',
    `Multiple wranglers (${names.join(', ')}) match the Account saved against this project: ${accountId}`
  );
} else {
  this.alertService.warn(
    'No Wranglers Match Account',
    `No wranglers match the Account saved against this project: ${accountId}`
  );
}
```